README.adoc

```diff
@@ -253,23 +211,95 @@ public class Album {
 }
 ----
 
-== Cloud Spanner Hibernate ORM Limitations
 
-The Cloud Spanner Hibernate Dialect supports most of the standard Hibernate and Java Persistence annotations, but there are important differences in features because of differences in https://cloud.google.com/spanner/docs/schema-and-data-model[Cloud Spanner's data model] from traditional SQL databases.
+=== Performance Optimizations
+
+There are some practices which can improve the execution time of your statements as well.
+
+==== Be Clear about Inserts or Updates
+
+Sometimes you may generate IDs for their entities in your code instead of relying on the `@GeneratedValue` annotation.
+
+```
+DbEntity entity = new DbEntity();
+entity.id = UUID.randomUUID();
+session.saveOrUpdate(entity);
+```
 
-The following sections below describe the feature set of the Cloud Spanner Hibernate Dialect as well as its limitations.
+This practice may result in Hibernate executing additional `SELECT` statements that you may not have expected because it will perform a `SELECT` on the table to determine if a record in the table already has the entity's ID. If so, it will perform an update, otherwise it will save a new entity.
+
+These extra `SELECT` statements could yield large performance issues because they may lock an entire table depending on how they are generated.
+
+In these cases, we recommend to either:
+
+* Let Hibernate generate the ID by leaving the entity's `id` null and annotate the field with `@GeneratedValue`. Hibernate will know that the record did not exist prior if it generates a new ID.
```

Comment:
This might be better in "use generated UUIDs" section, with this current section just linking to it for the first solution. And then this section would have only the use-persist-to-be-explicit-about-inserts.

Reply:
Done.